fastlane_core/lib/fastlane_core/crash_reporter/crash_reporter.rb

```diff
@@ -4,20 +4,27 @@
 module FastlaneCore
   class CrashReporter
     class << self
+      @did_report_crash = false
+
       def crash_report_path
         File.join(FastlaneCore.fastlane_user_dir, 'latest_crash.json')
       end
 
       def enabled?
-        !FastlaneCore::Env.truthy?("FASTLANE_OPT_OUT_CRASH_REPORTING")
+        !FastlaneCore::Env.truthy?("FASTLANE_OPT_OUT_CRASH_REPORTING") && !@did_report_crash
```

Comment:
I think putting `did_report_crash` into this method muddies the meaning of `enabled?` and it would be better to do this checking in `report_crash`.